spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/SparkShufflingDataRewritePlanner.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.actions;
+
+import java.util.Map;
+import java.util.Set;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.actions.BinPackRewriteFilePlanner;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.util.PropertyUtil;
+
+/**
+ * Extends the {@link BinPackRewriteFilePlanner} with the possibility to set the expected
```

Comment:
Needs more description here. Extends the BinPack rewriter for Rewriters that induce a distributed shuffle to reorganize data. (or something like this)

Reply:
Again, this is for the Planner. There was a specific configuration for the Shuffling rewriters which is modifying the plans. So I had to extend the core planners with the new configuration.

Alternatively we can move this functionality to the generic planner. In this case we don't need a specific class here. In this case the we have this config available for every planner (might be useful if there are compaction changes on any rewrite), but we have to reintroduce the flag `shufflingPlanner` (maybe with another name) to decide which Runner implementation to use.

Reply:
I was just talking about the Java Doc. I have no problem with the code organization but the key detail about this planner is that it produces plans for Shuffling Rewrites, not that it has an additional option.

Reply:
I thought this class adds an additional option of `COMPRESSION_FACTOR` used when calculating `expectedOutputFiles`

Reply:
Thanks for all the suggestions.
Updated the javadoc based on your input.